subsys/shell/modules/kernel_service.c

```diff
@@ -73,9 +74,19 @@ static void shell_tdata_dump(const struct k_thread *thread, void *user_data)
 		      thread,
 		      tname ? tname : "NA");
 	shell_fprintf((const struct shell *)user_data, SHELL_NORMAL,
-		      "\toptions: 0x%x, priority: %d\n",
+		      "\toptions: 0x%x, priority: %d timeout %d\n",
 		      thread->base.user_options,
-		      thread->base.prio);
+		      thread->base.prio,
+		      thread->base.timeout.dticks);
+	shell_fprintf((const struct shell *)user_data, SHELL_NORMAL,
+			"\tflags: %s%s%s%s%s%s%s\n",
+			thread->base.thread_state & _THREAD_DUMMY ? "dummy " : "",
```

Comment:
that would look nicer if you turn it to a function (e.g. `thread_state_to_str()`)

Reply:
yeah, make sense. Thank, you for this idea. Done